src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -2552,8 +2557,74 @@ void Lowering::ContainCheckNeg(GenTreeOp* neg)
 }
 
 //----------------------------------------------------------------------------------------------
-// Try converting SELECT/SELECTCC to CINC/CINCCC. Conversion is possible only if
-// both the trueVal and falseVal are integral constants and abs(trueVal - falseVal) = 1.
+// TryLowerCselToCinvOrCneg: Try converting SELECT/SELECTCC to SELECT_INV/SELECT_INVCC. Conversion is possible only if
+// one of the operands of the select node is inverted.
+//
+// Arguments:
+//     select - The select node that is now SELECT or SELECTCC
+//     cond   - The condition node that SELECT or SELECTCC uses
+//
+void Lowering::TryLowerCselToCinvOrCneg(GenTreeOp* select, GenTree* cond)
+{
+    assert(select->OperIs(GT_SELECT, GT_SELECTCC));
+
+    bool     shouldReverseCondition;
+    GenTree* invertedOrNegatedVal;
+    GenTree* nonInvertedOrNegatedVal;
+
+    GenTree*   trueVal  = select->gtOp1;
+    GenTree*   falseVal = select->gtOp2;
+    const bool isCneg   = ((trueVal->gtOper == GT_NEG) || (falseVal->gtOper == GT_NEG));
+
+    if ((trueVal->gtOper == GT_NOT) || (trueVal->gtOper == GT_NEG))
+    {
+        shouldReverseCondition  = true;
+        invertedOrNegatedVal    = trueVal->AsOp()->gtOp1;
+        nonInvertedOrNegatedVal = falseVal;
+        BlockRange().Remove(trueVal);
```

Comment:
This is not safe without an interference check due to how register lifetimes of local variables work. For example, it will be incorrect in the following case:

```scala
val = LCL_VAR V00
trueVal = NEG val
cns = CNS_INT 0
STORE_LCL_VAR(V00, cns)
result = SELECTCC(cond, trueVal, falseVal)
```
(legal IR)
->
```scala
val = LCL_VAR V00
cns = CNS_INT 0
STORE_LCL_VAR(V00, cns)
result = SELECT_NEGCC(cond, val, falseVal)
```
(illegal IR, uses of LCL_VARs happen at the user and no interfering stores are allowed)

So this needs an `IsInvariantInRange(trueVal->gtGetOp1(), select)`. Same for the other operand.

Reply:
Added a check for both operands.